processor/processor.go

```diff
@@ -2257,6 +2362,8 @@ func (proc *Handle) transformSrcDest(
 				reportMetrics = append(reportMetrics, successMetrics...)
 				reportMetrics = append(reportMetrics, failedMetrics...)
 				reportMetrics = append(reportMetrics, diffMetrics...)
+				reportMetrics = append(reportMetrics, filteredMetrics...)
+				reportMetrics = append(reportMetrics, addedMetrics...)
```

Comment:
AFAIK diff metrics were supposed to capture these kind of differences. @Sidddddarth do we need diff metrics anymore?

Reply:
considering that `filtered` means the same thing, `diff` can be removed.
Also, request to reconsider the name `filtered`, would `dropped` be better suited?
And events can be dropped in other areas of the processor as well(e.g.: event filter), not just user transforms, so this could be used at all three stages of the transform(UT, event filter, DT). In fact at the destination filter too.

Reply:
for the transformer report - we need the input and output numbers. `succeeded` gives output, input can be derived as `succeeded - diff` since we cannot always use `filtered` and `added` (input = succeeded - added +filtered) as they have nuances around transformBatch

Reply:
I am open to name change. Did not want to use `dropped` from product perspective as that has negative connotation when showing to user.